sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -356,6 +452,12 @@ private void processSendWork() {
                 delivery = sender.delivery(deliveryTag.getBytes(UTF_8));
                 delivery.setMessageFormat(workItem.getMessageFormat());
 
+                AmqpTransaction transactionId = workItem.getTransactionId();
```

Comment:
This can be even more abstracted by accepting a deliveryState parameter rather than an AmqpTransaction. That way, all usages of AmqpTransaction are removed from this class and abstracted into the coordinator. The RetriableWorkItem no longer has a reference on a transaction, and a delivery state instead.

```java
Mono<DeliveryState> send(byte[] bytes, int arrayOffset, int messageFormat, DeliveryState deliveryState) {
}
```

Reply:
We use AmqpSendLink directly in `ServiceBusSenderAsyncClient` to send message. This change will introduce a amqp deep concept of `DeliveryState ` into `ServiceBusSenderAsyncClient` which is unnecessary, I think .  `AmqpTransaction` is working as abstraction here.